analyses/gene-set-enrichment-analysis/01-conduct-gsea-analysis.R

```diff
@@ -0,0 +1,162 @@
+################################################################################
+
+
+
+#### Set Up Libraries --------------------------------------------------------------------
+
+## Load and/or install libraries ##
+library(tidyr)
+library(readr)
+library(tibble)
+library(optparse)
+
+# Magrittr pipe
+`%>%` <- dplyr::`%>%`
+
+if (!("msigdbr" %in% installed.packages())) {
+  install.packages("msigdbr")
+}
+if (!("BiocManager" %in% installed.packages())) {
+  install.packages("BiocManager")
+}
+library(BiocManager)
+if (!("GSVA" %in% installed.packages())) {
+  BiocManager::install("GSVA")
+}
+library(msigdbr) ## Contains the hallmark data sets
+library(GSVA)    ## Performs GSEA analysis
+
+
+#### Set Up optparse --------------------------------------------------------------------
+
+## Define arguments
+option_list <- list(
+  optparse::make_option(
+    c("--smallset"),
+    type = "logical",
+    default = FALSE,
+    help = "Set as TRUE to limit the number of samples evaluated, largely for running in CI."
+  ),
+    optparse::make_option(
+    c("--smallset_size"),
+    type = "double",
+    default = 15,
+    help = "Size of small sample, only used if argument `smallset` is specified as TRUE."
+  )
+)
+
+## Read in arguments
+opt_parser <- optparse::OptionParser(option_list = option_list)
+opt <- optparse::parse_args(opt_parser)
+
+
+## Sanity check the options. 
+MIN_SAMPLE_SIZE <- 15 # At least 15 should be run for small samples
+
+if (is.null(opt$smallset) | is.na(opt$smallset)){
+    opt$smallset <- FALSE
+}
+if (opt$smallset == 0) opt$smallset <- FALSE
+if (opt$smallset == 1) opt$smallset <- TRUE
+
+if (!(typeof(opt$smallset) == "logical"))  {
+  stop("ERROR: The optional argument --smallset must be a logical argument, TRUE (1) or FALSE (0).")
+}  
+
+if (opt$smallset == TRUE){
+  if (!(typeof(opt$smallset_size) == "double"))  {
+    stop("ERROR: The optional argument --smallset_size must be a number")
+  }
+  # Ensure a reasonable amount at least
+  if (opt$smallset_size < MIN_SAMPLE_SIZE) {
+    opt$smallset_size <- MIN_SAMPLE_SIZE
+  } 
+  ## Ceil the smallset value in case of decimal
+  opt$smallset_size <- ceiling(opt$smallset_size)
+}
+
+#### Set Up paths and file names --------------------------------------------------------------------
+
+## Define directories
+data_dir    <- file.path("..", "..", "data") 
+results_dir <- "results"
+if (!dir.exists(results_dir)) dir.create(results_dir)
+
+## Define input files
+## Define updated expression matrix. Columns are biospecimen ids and values are RSEM FPKM for stranded samples collapsed to gene symbol (gene-level)
+expression_data_file <- file.path(data_dir, "pbta-gene-expression-rsem-fpkm-collapsed.stranded.rds")
+
+## Define output file
+gsea_scores_output_file <- file.path(results_dir, "gsea_scores.tsv")
```

Comment:
```suggestion
gsea_scores_output_file <- file.path(results_dir, "gsva_scores.tsv")
```
may be a more appropriate file name.